### PR TITLE
Move `ResourceInformation` abstract base class to FWCore/AbstractServices, and few additional improvements

### LINKED ISSUE #1198: Move ResourceInformationService to FWCore/AbstractServices
Also

- [x] change `ResourceInformation::acceleratorTypes()` to hold the contents of `process.options.accelerators` as strings
- [x] Add `bool ResourceInformation::hasGpuNvidia()`

See https://github.com/cms-sw/cmssw/pull/47175#pullrequestreview-2576400675

### LINKED ISSUE #1200: Migrate code to use `ResourceInformation::hasNvidiaGpu()`
See https://github.com/cms-sw/cmssw/pull/47175#pullrequestreview-2576400675

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

This PR is part of https://github.com/cms-sw/cmssw/issues/30044.

It primarily moves the `ResourceInformation` abstract base class to a new `FWCore/AbstractServices` package, in order to (in a subsequent PR) to have the `ResourceInformation` to return `HardwareResourcesDescription` object (added in  https://github.com/cms-sw/cmssw/pull/47175, located in `DataFormats/Provenance`). We can't make `FWCore/Utilities` to depend on `DataFormats/Provenance`, and moving a service base class to `DataFormats/Provenance` didn't feel good either. We will later move other abstract base classes from `FWCore/Utilities` to `FWCore/AbstractServices`.

In addition, to prepare for `HardwareResourcesDescription`, this PR replaces the `enum` for "available accelerator types" as string(s) for "selected accelerators", that is just what comes from `process.options.accelerators`.

In addition (mostly to just take advantage of the PR touching these packages), this PR adds `hasGpuNvidia()` member function to `ResourceInformation` that can be used (e.g. in `TensorFlow` or `PyTorch` packages) to check if the job should use NVIDIA GPUs in a way that does not directly depend on CUDA. The last commit then changes the code in `TensorFlow` to use this function.

Resolves https://github.com/cms-sw/framework-team/issues/1198
Resolves https://github.com/cms-sw/framework-team/issues/1200

#### PR validation:

Code compiles

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

To be eventually backported to 15_0_X.

### CODE REVIEW

**Comment by makortel** [comment]
enable gpu

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
The update fixes the CPU test failure. The GPU test failure occurs already in IBs.

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
CPU differences are related to https://github.com/cms-sw/cmssw/issues/47071. GPU differences seem compatible with the non-reproducibilities in the pixel code.

**Comment by makortel** [comment]
+core

**Comment by makortel** [comment]
+heterogeneous

**Comment by makortel** [comment]
@cms-sw/ml-l2 Could you review and sign, please? Thanks!

**Comment by valsdav** [comment]
+ml

**Comment by mandrenguyen** [comment]
+1
